File: app/src/main/java/com/petal/browser/activity/Settings_Activity.java
package com.petal.browser.activity;

import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.WindowCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.core.view.WindowInsetsControllerCompat;

import com.petal.browser.R;
import com.petal.browser.compose.settings.PetalSettingsBridge;
import com.petal.browser.unit.BrowserUnit;
import com.petal.browser.unit.HelperUnit;

public class Settings_Activity extends AppCompatActivity {

    @Override
    protected void attachBaseContext(android.content.Context newBase) {
        super.attachBaseContext(HelperUnit.applyLanguage(newBase));
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        HelperUnit.initTheme(this);
        EdgeToEdge.enable(this);
        WindowInsetsControllerCompat controller = WindowCompat.getInsetsController(getWindow(), getWindow().getDecorView());
        controller.hide(WindowInsetsCompat.Type.systemBars());
        controller.setSystemBarsBehavior(WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
        setContentView(PetalSettingsBridge.createSettingsView(this, () -> {
            finish();
            return kotlin.Unit.INSTANCE;
        }));
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_help, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem menuItem) {
        if (menuItem.getItemId() == android.R.id.home) finish();
        else if (menuItem.getItemId() == R.id.menu_help) {
            Uri webpage = Uri.parse("https://github.com/shreyagarwal72/petal");
            BrowserUnit.intentURL(this, webpage);
        }
        return true;
    }
}

File: app/src/main/java/com/petal/browser/activity/Settings_Activity.kt
package com.petal.browser.activity

import android.os.Bundle
import androidx.activity.EdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import com.petal.browser.compose.settings.PetalSettingsBridge
import com.petal.browser.compose.settings.SettingsCategory
import com.petal.browser.unit.HelperUnit

class Settings_Activity : AppCompatActivity() {

    override fun attachBaseContext(newBase: android.content.Context) {
        super.attachBaseContext(HelperUnit.applyLanguage(newBase))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        HelperUnit.initTheme(this)
        EdgeToEdge.enable(this)
        val controller = WindowCompat.getInsetsController(window, window.decorView)
        controller.hide(WindowInsetsCompat.Type.systemBars())
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE

        var initialCategory = SettingsCategory.OVERVIEW
        val requestedCategory = intent?.getStringExtra(EXTRA_SETTINGS_CATEGORY)
        if (requestedCategory != null) {
            try {
                initialCategory = SettingsCategory.valueOf(requestedCategory)
            } catch (ignored: IllegalArgumentException) {
                // Unknown/typo'd category name - fall back to the overview rather than crash.
            }
        }

        setContentView(
            PetalSettingsBridge.createSettingsView(this, initialCategory) {
                finish()
            }
        )
    }

    companion object {
        /**
         * Optional String extra naming a [SettingsCategory] enum constant (e.g.
         * "API_INTEGRATIONS"). When present, Settings opens straight to that category's
         * page instead of the root category list.
         */
        const val EXTRA_SETTINGS_CATEGORY = "settings_category"
    }
}
